cocos/core/math/vec3.ts

```diff
@@ -1057,4 +1129,55 @@ export function v3 (x?: number | Vec3, y?: number, z?: number) {
     return new Vec3(x as any, y, z);
 }
 
+/**
+ * Chooses an arbitrary unit vector that is perpendicular to input.
+ */
+const chooseAnyPerpendicular = (() => {
```

Comment:
It is better to drop the minimal absolute component in function chooseAnyPerpendicular